indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi.cpp

```diff
@@ -1567,81 +1524,79 @@ static indigo_result focuser_detach(indigo_device *device) {
-            RPiCamera *pRPicamera = PRIVATE_DATA->pRPiCamera;
-            ControlList m_controls;
-            auto controlsInfo = pRPicamera->GetControls();
+			RPiCamera *pRPicamera = PRIVATE_DATA->pRPiCamera;
+			ControlList m_controls;
+			auto controlsInfo = pRPicamera->GetControls();
 
-            m_controls.set(controls::Contrast, X_CCD_CONTRAST_ITEM->number.value);
+			m_controls.set(controls::Contrast, X_CCD_CONTRAST_ITEM->number.value);
 			m_controls.set(controls::Saturation, X_CCD_SATURATION_ITEM->number.value);
-            m_controls.set(controls::Brightness, X_CCD_BRIGHTNESS_ITEM->number.value);
-
-            pRPicamera->Set(m_controls);
+			m_controls.set(controls::Brightness, X_CCD_BRIGHTNESS_ITEM->number.value);
 
-            indigo_update_property(device, X_CCD_ADVANCED_PROPERTY, NULL);
+			pRPicamera->Set(m_controls);
 
-        }
+			indigo_update_property(device, X_CCD_ADVANCED_PROPERTY, NULL);
+		}
 		return INDIGO_OK;
 	}
-    return indigo_ccd_change_property(device, client, property);
- }
- 
- static indigo_result ccd_detach(indigo_device *device) {
-     assert(device != NULL);
-     if (IS_CONNECTED) {
-         indigo_set_switch(CONNECTION_PROPERTY, CONNECTION_DISCONNECTED_ITEM, true);
-         ccd_connect_callback(device);
-
-    if (X_CCD_ADVANCED_PROPERTY) {
-		indigo_release_property(X_CCD_ADVANCED_PROPERTY);
-	}
-     }
-     indigo_release_property(RPI_AF_PROPERTY);
-
-     INDIGO_DEVICE_DETACH_LOG(DRIVER_NAME, device->name);
-     return indigo_ccd_detach(device);
- }
- 
- #define MAX_DEVICES                   10
- 
- static indigo_device *devices[MAX_DEVICES];
- static std::string new_eid;
- static pthread_mutex_t device_mutex = PTHREAD_MUTEX_INITIALIZER;
- 
- bool IsNew(std::string eid) {
-     for (int i = 0; i < MAX_DEVICES; i++) {
-         indigo_device *device = devices[i];
-         if (device && PRIVATE_DATA->eid == eid) {
-             return false;
-         }
-     }
-     return true;
- }
-
- std::string GetCameraName(std::string new_eid){
-
-    size_t lastSlash = new_eid.find_last_of('/'); 
-    // Extract the word after the last slash
-    //
-    std::string lastWord = (lastSlash != std::string::npos) ? new_eid.substr(lastSlash + 1) : new_eid;
-
-    return lastWord;
- }
+	return indigo_ccd_change_property(device, client, property);
+}
+
+static indigo_result ccd_detach(indigo_device *device) {
+	assert(device != NULL);
+	if (IS_CONNECTED) {
+		indigo_set_switch(CONNECTION_PROPERTY, CONNECTION_DISCONNECTED_ITEM, true);
+		ccd_connect_callback(device);
+
+		if (X_CCD_ADVANCED_PROPERTY) {
+			indigo_release_property(X_CCD_ADVANCED_PROPERTY);
+		}
+	}
+	indigo_release_property(RPI_AF_PROPERTY);
+
+	INDIGO_DEVICE_DETACH_LOG(DRIVER_NAME, device->name);
+	return indigo_ccd_detach(device);
+}
+
+#define MAX_DEVICES 10
+
+static indigo_device *devices[MAX_DEVICES];
+static std::string new_eid;
+static pthread_mutex_t device_mutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
Yet another POSIX mutex... 5 mutexes for a simple camera driver is an overkill and a recipe for disaster.
Also mutex usage is inconsistent - you are mixing std::mutex and pthread_mutex_t. I would strongly suggest that you revise the mutex usage and thread synchronization. I have a strong suspicion that broken frames are a result of broken thread synchronization.